File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> res;

        int num_zeroes = 0;
        int product = 1;
        for (int n : nums) {
            if (n == 0) {
                num_zeroes++;
            } else {
                product *= n;
            }
        }

        for (int n : nums) {
            if (n != 0) {
                if (num_zeroes) res.push_back(0);
                else res.push_back(product / n);
            } else {
                if (num_zeroes > 1) res.push_back(0);
                else res.push_back(product);
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
import heapq

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        map = {}
        for num in nums:
            if num in map:
                map[num] += 1
            else:
                map[num] = 1

        pq = []
        for num in map.keys():
            heapq.heappush(pq, (-1 * map[num], num))

        res = []
        
        for i in range(k):
            res.append(heapq.heappop(pq)[1])

        return res

        
        